egs/madcat_ar/v1/local/create_line_image_from_page_image.py

```diff
@@ -0,0 +1,535 @@
+
+    Returns
+    ------- 
+    [(int, int)]: 4 corner points of rectangle.
+    """
+    p1, p2, p3, p4 = bounding_box.corner_points
+    x1, y1 = p1
+    x2, y2 = p2
+    x3, y3 = p3
+    x4, y4 = p4
+    center_x, center_y = center
+    rotation_angle_in_rad = -get_smaller_angle(bounding_box)
+    x_dash_1 = (x1 - center_x) * cos(rotation_angle_in_rad) - (y1 - center_y) * sin(rotation_angle_in_rad) + center_x
+    x_dash_2 = (x2 - center_x) * cos(rotation_angle_in_rad) - (y2 - center_y) * sin(rotation_angle_in_rad) + center_x
+    x_dash_3 = (x3 - center_x) * cos(rotation_angle_in_rad) - (y3 - center_y) * sin(rotation_angle_in_rad) + center_x
+    x_dash_4 = (x4 - center_x) * cos(rotation_angle_in_rad) - (y4 - center_y) * sin(rotation_angle_in_rad) + center_x
+
+    y_dash_1 = (y1 - center_y) * cos(rotation_angle_in_rad) + (x1 - center_x) * sin(rotation_angle_in_rad) + center_y
+    y_dash_2 = (y2 - center_y) * cos(rotation_angle_in_rad) + (x2 - center_x) * sin(rotation_angle_in_rad) + center_y
+    y_dash_3 = (y3 - center_y) * cos(rotation_angle_in_rad) + (x3 - center_x) * sin(rotation_angle_in_rad) + center_y
+    y_dash_4 = (y4 - center_y) * cos(rotation_angle_in_rad) + (x4 - center_x) * sin(rotation_angle_in_rad) + center_y
+    return x_dash_1, y_dash_1, x_dash_2, y_dash_2, x_dash_3, y_dash_3, x_dash_4, y_dash_4
+
+
+def pad_image(image):
+    """ Given an image, returns a padded image around the border.
+        This routine save the code from crashing if bounding boxes that are
+        slightly outside the page boundary.
+
+    Returns
+    -------
+    image: page image
+    """
+    padded_image = Image.new('RGB', (image.size[0] + padding, image.size[1] + padding), "white")
+    padded_image.paste(im=image, box=(offset, offset))
+    return padded_image
+
+
+def update_minimum_bounding_box_input(bounding_box_input):
+    """ Given list of 2D points, returns list of 2D points shifted by an offset.
+
+    Returns
+    ------
+    points [(float, float)]: points, a list or tuple of 2D coordinates
+    """
+    updated_minimum_bounding_box_input = []
+    for point in bounding_box_input:
+        x, y = point
+        new_x = x + offset
+        new_y = y + offset
+        word_coordinate = (new_x, new_y)
+        updated_minimum_bounding_box_input.append(word_coordinate)
+
+    return updated_minimum_bounding_box_input
+
+
+def set_line_image_data(image, line_id, image_file_name):
+    """ Given an image, saves a flipped line image. Line image file name
+        is formed by appending the line id at the end page image name.
+    """
+
+    base_name = os.path.splitext(os.path.basename(image_file_name))[0]
+    image_file_name_wo_tif, b = image_file_name.split('.tif')
+    line_id = '_' + line_id.zfill(4)
+    line_image_file_name = base_name + line_id + '.tif'
+    image_path = os.path.join(output_directory, line_image_file_name)
+    imgray = image.convert('L')
+    imgray_rev_arr = np.fliplr(imgray)
```

Comment:
This line flips the image. It was done for Arabic data. Please remove this line.

Reply:
The documentation also says it flips, presuably that should be fixed too.  But I don't see how it could have worked if the images were all backwards.